lib/src/main/java/se/gustavkarlsson/track/sqlite/CloseableRecordCursorSequence.kt

```diff
@@ -8,15 +8,12 @@ internal class CloseableRecordCursorSequence(
 ) : CloseableSequence<Record> {
     private val sequence = sequence {
         cursor.use { cursor ->
-            var hasData = cursor.checkForDataOrClose()
-            while (hasData) {
-                val record = cursor.readExistingRecord()
-                hasData = cursor.checkForDataOrClose()
-                yield(record)
-            }
+            while (cursor.hasData) yield(cursor.readExistingRecord())
         }
     }
 
+    private val RecordCursor.hasData get() = checkForDataOrClose()
```

Comment:
To be honest, this extension function is not really necessary. Calling `checkForDataOrClose()` directly should be fine....